packages/react/src/themes/teams/components/Animation/animationStyles.ts

```diff
@@ -1,5 +1,23 @@
+import { AnimationProp, ProviderContextPrepared } from '../../../types'
+import createAnimationStyles from '../../../../lib/createAnimationStyles'
+
 export default {
   root: () => ({
     display: 'inline-block',
   }),
+  children: ({ props, variables, ...context }) => {
+    const animation: AnimationProp = {
+      name: props.name,
+      keyframeParams: props.keyframeParams,
+      duration: props.duration,
+      delay: props.delay,
+      iterationCount: props.iterationCount,
+      direction: props.direction,
+      fillMode: props.fillMode,
+      playState: props.playState,
+      timingFunction: props.timingFunction,
+    }
+
+    return createAnimationStyles(animation, context as ProviderContextPrepared)
```

Comment:
lets discuss this - it seems that we could simplify this a bit and avoid leaking `ProviderContextPrepared` to styles API _for all components_ - this is, essentially, the thing I would like to prevent. Essentially, we would like to pass (for now) only theme object, with nothing else being leaked